packages/mcp-client/src/McpClient.ts

```diff
@@ -170,7 +198,7 @@ export class McpClient {
 			const client = this.clients.get(toolName);
 			if (client) {
 				const result = await client.callTool({ name: toolName, arguments: toolArgs, signal: opts.abortSignal });
-				toolMessage.content = (result.content as Array<{ text: string }>)[0].text;
+				toolMessage.content = ResultFormatter.format(result as CallToolResult);
```

Comment:
(nit) i take this as a "bug" in the TS MCP SDK that we need to cast this as a CallToolResult, no?